lib/sax.js

```diff
@@ -554,6 +554,28 @@ function parseDCC(source,start,domBuilder,errorHandler){//sure start with '<!'
 			domBuilder.endCDATA() 
 			return end+3;
 		}
+		// TODO: move to parseInternalDTD (see below)
+		if(source.substr(start+2,6) == 'ENTITY'){
+			var end = source.indexOf('>', start+8);
+			var chunk = source.substring(start+8, end)
+			var match = chunk.match(/\s+(\w+)\s+(["'])(.*)/);
+			if (!match) {
+				errorHandler.error("Malformed internal entity declaration");
+				return -1;
+			}
+			var name = match[1];
+			var delim = match[2]
+			var rest = match[3];
+			var value = rest.substring(0, rest.indexOf(delim));
+			domBuilder.internalEntityDecl(name, value);
+			let nextTagStart = source.indexOf('<', end);
+			let dtdEnd = source.indexOf(']>', end);
```

Comment:
To fix the lint check you need to stick to ES5 here.
```suggestion
			var nextTagStart = source.indexOf('<', end);
			var dtdEnd = source.indexOf(']>', end);
```